mypy/build.py

```diff
@@ -891,7 +891,12 @@ def write_cache(id: str, path: str, tree: MypyFile,
         manager.trace("Interface for {} has changed".format(id))
 
     # Obtain and set up metadata
-    st = manager.get_stat(path)  # TODO: Handle errors
+    try:
+        st = manager.get_stat(path)
+    except OSError:
+        manager.log("Cannot write {}".format(path))
```

Comment:
1. We should try to delete the nonced file here (and the data file; it's invalidated by the mismatched `data_mtime` anyway).
2. Would also be good to at least stringify the OSError to get an idea why the stat failed.
3. The error message is misleading. It's a source stat issue, not a cache write issue.